media/js/mozorg/contribute/taskview.js

```diff
@@ -0,0 +1,88 @@
+$(function() {
+    'use strict';
+
+    var $stepOne = $('#step_one');
+    var $thankYou = $('#thankyou');
+    var $getFirefox = $('.get-firefox');
+    var $downloadButton = $('.download-link');
+
+    var $followButton = $('.follow-mozilla')
+    var intentURL = $followButton.attr('href');
+
+    // some tasks, like install Whimsy, required the user to be using Firefox
+    if ($getFirefox.length > -1 && !isFirefox()) {
+        $getFirefox.toggleClass('hidden');
+    }
+
+    /**
+     *
+     */
+    function handleVisibilityChange() {
+        document.addEventListener('visibilitychange', function() {
+            // we wait until our current tab is visible before
+            // showing the thank you message.
+            if (document.visibilityState === 'visible') {
+                // toggles number to check mark display
+                $stepOne.addClass('completed');
+                taskComplete();
```

Comment:
Should we remove the `visibilitychange` event listener after this fires? It only has to execute once, right?